fabric-tree-api-v1/src/main/java/net/fabricmc/fabric/api/tree/v1/SimpleSaplingBlock.java

```diff
@@ -0,0 +1,23 @@
+package net.fabricmc.fabric.api.tree.v1;
+
+import net.minecraft.block.SaplingBlock;
+import net.minecraft.block.sapling.SaplingGenerator;
+
+/**
+ * <p>A {@link SaplingBlock} with a public constructor to create your own saplings.</p>
+ *
+ * <p>This could be done using an access widener on the {@link SaplingBlock} constructor, but access wideners should be avoided where possible.</p>
+ *
+ * <p>
+ *     An example of using this implementation:<br>
+ *
+ *     <code>
+ *			public static final SimpleSaplingBlock MY_SAPLING = new SimpleSaplingBlock(new SimpleSaplingGenerator(MY_TREE_CONFIGURED), FabricBlockSettings.copyOf(Blocks.OAK_SAPLING));
+ *     </code>
+ * </p>
+ */
+public class SimpleSaplingBlock extends SaplingBlock {
+	public SimpleSaplingBlock(SaplingGenerator generator, Settings settings) {
```

Comment:
Instead of creating this class, adding an access widener entry to make sapling block's constructor public is better.

In addition, users can always create a `new SaplingBlock(generator, settings) {}` to call the protected constructor.

Reply:
Access wideners aren't exposed, though. Only your own AW will be applied at compile time (and deps' at runtime).

Reply:
Yeah, IDEs don't support AWs very well, which is its biggest disadvantage.

Reply:
I've readded this to the PR because AWs aren't exposed. It'll stay as a temporary solution